src/functTest/groovy/com/bmuschko/gradle/docker/tasks/container/DockerCreateContainerFunctionalTest.groovy

```diff
@@ -16,6 +16,7 @@
 package com.bmuschko.gradle.docker.tasks.container
 
 import com.bmuschko.gradle.docker.AbstractFunctionalTest
+import org.gradle.api.tasks.TaskExecutionException
 import org.gradle.testkit.runner.BuildResult
 
 class DockerCreateContainerFunctionalTest extends AbstractFunctionalTest {
```

Comment:
@cdancy In general would should externalize some of the fixture logic. This gets copied and over and over again.

Reply:
I committed a change that simplifies the logic in this test class.

Reply:
I wonder if we could do the same but in a more global manner so that all tests clean up after themselves. That way no one will have to worry about tearing down a container it will simply just happen as part of setting up the test.

Reply:
Is that #383 ?

Reply:
I guess we could but it would require specific naming standards when creating images and containers so we can identify what to prune. We should also keep the base images. Those can be kept in between builds so we don't have to re-download them every time.

I could image it to work the following way:

1. When creating images in a functional test assign the label `gradle-docker-plugin`.
2. When running containers in a functional test assign the label `gradle-docker-plugin`.
3. Issue a `docker container prune --filter="label=gradle-docker-plugin"` and `docker image prune --filter="label=gradle-docker-plugin"` command in a `finalizedBy` task for `functionalTest`. I don't think we need to run this through the Docker Java API as it is more of a build management functionality.